File: src/main/java/com/kishlaly/ta/analyze/testing/sl/StopLossStrategy.java
package com.kishlaly.ta.analyze.testing.sl;

import com.kishlaly.ta.model.SymbolData;

public abstract class StopLossStrategy {

    protected Object config;
    protected boolean isVolatile;

    public StopLossStrategy(final Object config, boolean isVolatile) {
        this.config = config;
        this.isVolatile = isVolatile;
    }

    public void setConfig(Object config) {
        this.config = config;
    }

    public Object getConfig() {
        return this.config;
    }

    public boolean isVolatile() {
        return this.isVolatile;
    }

    public abstract double calculate(SymbolData data, int signalIndex);

    public abstract String printConfig();

    public String toString() {
        return getClass().getSimpleName() + ", " + printConfig();
    }

}

File: src/main/kotlin/com/kishlaly/ta/analyze/testing/sl/StopLossStrategy.kt
package com.kishlaly.ta.analyze.testing.sl

import com.kishlaly.ta.model.SymbolData

abstract class StopLossStrategy(
    val config: Any? = null,
    val isVolatile: Boolean = false
) {

    abstract fun calculate(data: SymbolData, signalIndex: Int): Double

    abstract override fun toString(): String

}